xilem/src/view/board/style_modifier.rs

```diff
@@ -0,0 +1,304 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::marker::PhantomData;
+
+use masonry::{widget::KurboShape, Affine};
+use vello::{
+    kurbo,
+    peniko::{self, Brush},
+};
+use xilem_core::{AnyView, DynMessage, MessageResult, Mut, View, ViewId, ViewMarker};
+
+use crate::{Pod, ViewCtx};
+
+use super::{AnyBoardView, BoardElement, GraphicsView};
+
+pub struct Transform<V, State, Action> {
+    child: V,
+    transform: Affine,
+    phantom: PhantomData<fn() -> (State, Action)>,
+}
+
+pub struct Fill<V, State, Action> {
+    child: V,
+    mode: peniko::Fill,
+    brush: Brush,
+    brush_transform: Option<Affine>,
+    phantom: PhantomData<fn() -> (State, Action)>,
+}
+
+pub struct Stroke<V, State, Action> {
+    child: V,
+    style: kurbo::Stroke,
+    brush: Brush,
+    brush_transform: Option<Affine>,
+    phantom: PhantomData<fn() -> (State, Action)>,
+}
+
+pub fn transform<State, Action, V>(child: V, transform: Affine) -> Transform<V, State, Action> {
+    Transform {
+        child,
+        transform,
+        phantom: PhantomData,
+    }
+}
+
+pub fn fill<State, Action, V>(child: V, brush: impl Into<Brush>) -> Fill<V, State, Action> {
+    Fill {
+        child,
+        mode: peniko::Fill::NonZero,
+        brush: brush.into(),
+        brush_transform: None,
+        phantom: PhantomData,
+    }
+}
+
+pub fn stroke<State, Action, V>(
+    child: V,
+    brush: impl Into<Brush>,
+    style: kurbo::Stroke,
+) -> Stroke<V, State, Action> {
+    Stroke {
+        child,
+        style,
+        brush: brush.into(),
+        brush_transform: None,
+        phantom: PhantomData,
+    }
+}
+
+impl<V, State, Action> Fill<V, State, Action> {
+    pub fn mode(mut self, mode: peniko::Fill) -> Self {
+        self.mode = mode;
+        self
+    }
+
+    pub fn brush_transform(mut self, brush_transform: Affine) -> Self {
+        self.brush_transform = Some(brush_transform);
+        self
+    }
+}
+
+impl<V, State, Action> Stroke<V, State, Action> {
+    pub fn brush_transform(mut self, brush_transform: Affine) -> Self {
+        self.brush_transform = Some(brush_transform);
+        self
+    }
+}
+
+pub trait GraphicsExt<State, Action>: GraphicsView<State, Action> + Sized {
```

Comment:
I think this is not accurate anymore, since e.g. a `PositionedView` is a `GraphicsView` as well.

So I think there are two ways to solve this and I don't have a strong opinion here either, IME the first option plays better with rust-analyzer though:

1. Separate modifier traits with a super element bound on a class of elements or one element, so something like this (similar as `xilem_web::interfaces::*`):
```rust
pub trait KurboShapeExt<State, Action>: GraphicsView<State, Action, Widget = KurboShape> + Sized {...}
```
2. As Olivier called it "Warehouse" traits, so something like:
```rust
pub trait SvgExt<State, Action>: GraphicsView<State, Action> + Sized {
    fn transform(self, affine: Affine) -> Transform<Self, State, Action>
    where
         // potentially in the future some kind of transform trait bound, maybe it's also general enough to be directly in the `SvgElement` trait
        Self::Element = KurboShape
    {
        transform(self, affine)
    }
    ...
}
```